modules/effects/src/main/scala/dev/profunktor/redis4cats/algebra/scripts.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018-2019 ProfunKtor
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.profunktor.redis4cats.algebra
+
+import dev.profunktor.redis4cats.effects.ScriptOutputType
+
+trait ScriptCommands[F[_], K, V] extends Scripting[F, K, V]
+
+trait Scripting[F[_], K, V] {
+  def eval(script: String, returnType: ScriptOutputType, keys: K*): F[returnType.Return[V]]
+  def eval(script: String, returnType: ScriptOutputType, keys: List[K], values: V*)(
+      implicit ev: K <:< Object
+  ): F[returnType.Return[V]]
+  def evalSha(script: String, returnType: ScriptOutputType, keys: K*): F[returnType.Return[V]]
+  def evalSha(script: String, returnType: ScriptOutputType, keys: List[K], values: V*)(
+      implicit ev: K <:< Object
+  ): F[returnType.Return[V]]
+  // This unfortunately has to take a V instead of String due to a bug in lettuce:
+  // https://github.com/lettuce-io/lettuce-core/issues/1010
+  def scriptLoad(script: V): F[String]
```

Comment:
We could alternatively leave out `scriptLoad` (and by that, basically `evalSha`) until Lettuce fixes this, but I feel it's better to have it, I assume most users would anyway have `V=String` when doing scripts.

Reply:
Yeah that's fine, let's keep it as is until the bug is fixed upstream.